.github/scripts/setup_runners.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+set -e
+
+#####################################
+# Variables / Parameters / Settings #
+#####################################
+
+# This script allows users to manually assign parameters
+if [ "$#" -ne 3 ] || [ "$1" = "--help" ] || [ "$1" = "-h" ]
+then
+  echo "Please assign at 3 parameters when running this script"
+  echo "Example: $0 \$ACTION \$EC2_INSTANCE_NAMES(,) \$GITHUB_TOKEN"
+  echo "Example: $0 \"run\" \"odfe-rpm-ism,odfe-rpm-sql\" \"<GitHub PAT>\""
+  exit 1
+fi
+
+SETUP_ACTION=$1
+SETUP_INSTANCE=`echo $2 | sed 's/,/ /g'`
+SETUP_TOKEN=$3
+SETUP_AMI_ID="ami-0397210a1b7e7e74c"
+SETUP_AMI_USER="ec2-user"
+SETUP_INSTANCE_TYPE="m5.xlarge"
+SETUP_INSTANCE_SIZE=20 #GiB
+SETUP_KEYNAME="odfe-release-runner"
+SETUP_SECURITY_GROUP="odfe-release-runner"
+SETUP_IAM_NAME="odfe-release-runner"
+GIT_URL_API="https://api.github.com/repos"
+GIT_URL_BASE="https://github.com"
+GIT_URL_REPO="opendistro-for-elasticsearch/opendistro-build"
+
+echo "###############################################"
+echo "Start Running $0 $1 $2"
+echo "###############################################"
+
+###############################################
+# Run / Start instances and bootstrap runners #
+###############################################
+if [ "$SETUP_ACTION" = "run" ]
+then
+
+  echo ""
+  echo "Run / Start instances and bootstrap runners [${SETUP_INSTANCE}]"
+  echo ""
+
+  # Provision VMs
+  for instance_name1 in $SETUP_INSTANCE
+  do
+    echo "[${instance_name1}]: Start provisioning vm"
+    aws ec2 run-instances --image-id $SETUP_AMI_ID --count 1 --instance-type $SETUP_INSTANCE_TYPE \
+                          --block-device-mapping DeviceName=/dev/xvda,Ebs={VolumeSize=$SETUP_INSTANCE_SIZE} \
+                          --key-name $SETUP_KEYNAME --security-groups $SETUP_SECURITY_GROUP \
+                          --iam-instance-profile Name=$SETUP_IAM_NAME \
+                          --tag-specifications "ResourceType=instance,Tags=[{Key=Name,Value=$instance_name1}]" > /dev/null 2>&1; echo $?
+    sleep 1
+  done
+
+  echo ""
+  echo "Sleep for 120 seconds for EC2 instances to start running"
```

Comment:
could you use some ec2 command to check the status?

Reply:
No really need to, if a EC2 VM just stuck in a loop, it will block the whole thing.
If one didnt start, others can still keep going, they will get removed / cleaned up anyway.
If anything act abnormally, just restart workflow.